### PR TITLE
Feat/assistant

### PR DESCRIPTION
**Pull request by MasterKenth**
# Assistant Templates

Assistant templates move configuration of "assistants" into the database.

An Assistant is a self-contained configuration of one or multiple streams with metadata. Uploaded files can be associated with an assistant to enable RAG features. Each individual stream can be separately configured (LLM provider etc.).

The existing planning stream has been converted to an Assistant (but requires a file collection id before it can be used - pick one from `/api/vector/list_collections` and edit assistant entry manually in database project for now).

Several example assistants are provided as well. Any assistant json file in `./data/assistant-templates` will be added during setup (if no project already exists).

## CRUD

CRUD API is added to manage assistants. This should be connected to a frontend in a future push. See backend `/docs` endpoint for more info.

## Misc

* Chat frontend has been updated to use the new `/assistant-stream` endpoint and list available assistants.
* Files associated with previous solution have been moved/removed from `fai_backend/llm/`.

### CODE REVIEW

File: data/assistant-templates/example.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+  "id": "example",
+  "name": "Example Assistant",
+  "description": "This is an *example assistant* for testing purposes.",
+  "sample_questions": [
```
Comment (dannil76) [inline data/assistant-templates/example.json:5]: Detalj, men kanske ska vara camelCase?

File: fai-rag-app/fai-backend/fai_backend/assistant/protocol.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        a stream of tokens (strings) of the response given by the LLM.
+        """
+        ...
+
+
+class IAssistantStreamProtocol(Protocol):
+    async def create_stream(self, stream_def: LLMStreamDef, get_vars: Callable[[], dict]) -> Stream[str, str]:
+        """
+
+        """
```
Comment (dannil76) [inline fai-rag-app/fai-backend/fai_backend/assistant/protocol.py:21]: kan skrivas med tre punkter
...

**Review by dannil76 — COMMENTED** [review]
Oklart på vilken nivå granskningen ska göras? Här får du min input som du får göra vad du vill med... :)

File: data/assistant-templates/example.json (hunk 1 above)
Comment (MasterKenth) [reply data/assistant-templates/example.json:5]: mjo, men den läser in rakt av genom pydantic och vill inte ha extra `sampleQuestions` -> `sample_questions` konvertering...jsonen används bara en gång vid setup också.

**Review by dannil76 — APPROVED** [review]

**Pull request merged**